services/pom.xml

```diff
@@ -36,6 +36,9 @@
         <maven.compiler.release>17</maven.compiler.release>
         <maven.compiler.source>17</maven.compiler.source>
         <maven.compiler.target>17</maven.compiler.target>
+        <mockito.version>1.9.5</mockito.version>
```

Comment:
AFAIK there were some voice in the past against introducing mockito to the Keycloak project. I can doublecheck with the team again, but on the other hand, it might be better to test with the real session?

So the alternative option is to avoid using tests directly inside `keycloak-services` module, but add the tests to the `testsuite/integration-arquillian` . That module is dedicated for integration tests, but it is also possible to somehow simulate unit testing, but with real `KeycloakSession` provided by real Keycloak server.

For example in the test `UserProfileTest` you can see pattern like this:
```
    @Test
    @ModelTest
    public void testRequiredByOptionalClientScope(KeycloakSession session) {
```

or the alternative pattern like this in some other methods:
```
    @Test
    public void testDoNotRemoveAttributes() {
        getTestingClient().server(TEST_REALM_NAME).run((RunOnServer) UserProfileTest::testDoNotRemoveAttributes);
    }

    private static void testDoNotRemoveAttributes(KeycloakSession session) {
       ... test something with KeycloakSession here ...
```

Is it possible to move the tests to `integration-arquillian` and use something like this?

Reply:
I've seen mockito in other modules(e.g. operator, oidc/spring-boot2, oidc/spring-security), thus I thought its ok. I'll check if moving it to integration-arquillian works.

Reply:
The integration-arquillian are based on junit4. I removed the mockito dependency and added "manual" mocks(in [9cb295c](https://github.com/keycloak/keycloak/pull/26484/commits/9cb295cee525ddf88ffc5a9fdd4d624fb2e567ec)).

Reply:
@wistefan I see, but it will be good to stick to junit4 and integration-arquillian instead of introducing additional "mock" classes. I have a feedback that there is a plan for some testsuite refactoring in KEycloak and having more various approaches to test things complicates for future refactoring of the tests. So will be nice to stick with what existing tests are doing instead of introducing new mechanisms and mocks at this stage...

I suggest to:
- Move tests to integration-arquillian and rely on the mechanisms like `UserProfileTest` is doing to test with Keycloak sessions (see my comment above). This has advantage that it tests with real KeycloakSession instead of mocks and can be good anyway as we would need integration tests at some stage anyway for oid4vc. The disadvantage is bigger time needed to run the test as it needs Keycloak server bootstrap. But once server is bootstrapped, it should be quite quick though.

- Possibly add some new package like `org.keycloak.testsuite.oid4vc` and add possibly something like `oid4vc,6` to this file https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/testsuites/base-suite (it will tell to use group-6 for oid4vc tests where are also OIDC tests and which does not take so long. Those groups are used for parallel running of testsuite in GH actions)

- Use some pattern like this to test with various wrappers (I agree it is not so nice, but should be sufficient and we use it in some other places like for example `FAPI2Test`):
```
@Test
public void testRsaKeys() {  
  testImpl(getRsaKey("my-rsa-key"), ... other args here ...);
}

@Test
public void testEcKeys() {
   testImpl(getECKey("my-ec-key") .. other args here ...);
}

private testImpl(KeyWrapper key, .. other args here ...) {
  .. test impl ...
}
```

Does it work for you?

Reply:
@mposolda I will change the tests as suggested. It would work for the tests in the current PR, however I fear that the tests for the issuer-endpoint(that I prepare for #25940 ) will become quite unreadable with the junit4 approach to parameterized tests.

Reply:
I moved the tests in [09d72dd](https://github.com/keycloak/keycloak/pull/26484/commits/09d72dd81e297f814456f38bb1ec04c17ab95a3f)

Reply:
Thanks!